etc/testing/circle/workloads/go-test-results/egress/main.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgconn"
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/jmoiron/sqlx"
+	gotestresults "github.com/pachyderm/pachyderm/v2/etc/testing/circle/workloads/go-test-results"
+	"github.com/pkg/errors"
+)
+
+var logger = log.Default()
+var postgresqlUser = os.Getenv("POSTGRESQL_USER")
+var postgresqlPassword = os.Getenv("POSTGRESQL_PASSWORD")
+var postgresqlHost = os.Getenv("POSTGRESQL_HOST")
+
+const (
+	jobInfoFileName string = "JobInfo.json"
+)
+
+type TestResultLine struct {
+	Time     time.Time `json:"Time"`
+	Action   string    `json:"Action"`
+	TestName string    `json:"Test"`
+	Package  string    `json:"Package"`
+	Elapsed  float32   `json:"Elapsed"`
+	Output   string    `json:"Output"`
+}
+
+// This is built and runs in a pachyderm pipeline to egress data uploaded to pachyderm to postgresql.
+func main() {
+	logger.Println("Beginning egress transform of data to sql DB")
+	inputFolder := os.Args[1]
+
+	db, err := sqlx.Open("pgx", fmt.Sprintf("user=%s password=%s host=%s port=5432 database=ci_metrics", postgresqlUser, postgresqlPassword, postgresqlHost))
```

Comment:
This is likely to be somewhat limiting if you ever want to set raw pgx options.  Check out the "fake" DSN we create in dbutil for how to interface sqlx and pgx: https://github.com/pachyderm/pachyderm/blob/master/src/internal/dbutil/db.go#L102